Plugins/GitUIPluginInterfaces/ObjectId.cs

```diff
@@ -320,27 +301,47 @@ public unsafe string ToShortString(int length = 8)
             char* buffer = stackalloc char[Sha1CharCount];
             char* p = buffer;
 
-            Write(_i1);
-            Write(_i2);
-            Write(_i3);
-            Write(_i4);
-            Write(_i5);
+            Write64(_i1);
+            Write64(_i2);
+            Write32(_i3);
 
             return new string(buffer, 0, length);
 
-            void Write(uint i)
+            void Write64(ulong i)
             {
-                *p++ = ParseHexDigit(i >> 28);
+                *p++ = ParseHexDigit(i >> 60);
+                *p++ = ParseHexDigit((i >> 56) & 0xF);
+                *p++ = ParseHexDigit((i >> 52) & 0xF);
+                *p++ = ParseHexDigit((i >> 48) & 0xF);
+                *p++ = ParseHexDigit((i >> 44) & 0xF);
+                *p++ = ParseHexDigit((i >> 40) & 0xF);
+                *p++ = ParseHexDigit((i >> 36) & 0xF);
+                *p++ = ParseHexDigit((i >> 32) & 0xF);
+                *p++ = ParseHexDigit((i >> 28) & 0xF);
                 *p++ = ParseHexDigit((i >> 24) & 0xF);
                 *p++ = ParseHexDigit((i >> 20) & 0xF);
                 *p++ = ParseHexDigit((i >> 16) & 0xF);
                 *p++ = ParseHexDigit((i >> 12) & 0xF);
                 *p++ = ParseHexDigit((i >> 8) & 0xF);
                 *p++ = ParseHexDigit((i >> 4) & 0xF);
                 *p++ = ParseHexDigit(i & 0xF);
+
+                char ParseHexDigit(ulong j) => j < 10 ? (char)('0' + j) : (char)(j + 0x57);
             }
 
-            char ParseHexDigit(uint j) => j < 10 ? (char)('0' + j) : (char)(j + 0x57);
+            void Write32(uint i)
+            {
+                *p++ = ParseHexDigit(i >> 28);
+                *p++ = ParseHexDigit((i >> 24) & 0xF);
+                *p++ = ParseHexDigit((i >> 20) & 0xF);
+                *p++ = ParseHexDigit((i >> 16) & 0xF);
+                *p++ = ParseHexDigit((i >> 12) & 0xF);
+                *p++ = ParseHexDigit((i >> 8) & 0xF);
+                *p++ = ParseHexDigit((i >> 4) & 0xF);
+                *p++ = ParseHexDigit(i & 0xF);
+
+                char ParseHexDigit(uint j) => j < 10 ? (char)('0' + j) : (char)(j + 0x57);
```

Comment:
```suggestion
                return;

                static char GetHexChar(uint nibble) => nibble < 10 ? (char)('0' + nibble) : (char)(('a' - 10) + nibble);
```

Reply:
Will be replaced by #11224 anyway - but only if #11224 takes comparable runtime. Does it?

Reply:
100K run of ToString() requires 13ms so less than a micro second (all are the same in the test though but do not expect much more). An additional allocation too. So nothing, even less than using LYNQ compared to direct. (It is the operations that is done 300K reading loading a repo that matters.)

(I had first planned to optimize that method but there is no benefit, just more code. It is too complicated with this PR, but if the others are merged first, we will not see it.)